src/compile/data/assemble.ts

```diff
@@ -307,8 +312,8 @@ export function assembleFacetData(root: FacetNode): VgData[] {
  * @param  data array
  * @return modified data array
  */
-export function assembleData(dataCompomponent: DataComponent): VgData[] {
-  let roots: SourceNode[] = vals(dataCompomponent.sources);
+export function assembleData(model: Model, dataComponent: DataComponent): VgData[] {
```

Comment:
> Would you rather the selection component be passed to assembleData

Yes, or just pass the whole model. But again, I am opposed to adding dataflow nodes in assemble. 

Can we add a dataflow node for selections that assembled the required transforms? We would add a node for each selection and then only actually insert the transform if it is needed.